src/pages/dashboard/components/Search.tsx

```diff
@@ -267,68 +417,149 @@ export function Search$() {
+                      'recurring_invoices',
+                      groups.recurring_invoices.length > 0
+                    )}
+                    {groups.recurring_invoices.map((entry) => {
+                      const index = currentIndex++;
+                      return renderSearchItem(entry, index);
+                    })}
+
+                    {renderGroupTitle('payments', groups.payments.length > 0)}
+                    {groups.payments.map((entry) => {
+                      const index = currentIndex++;
+                      return renderSearchItem(entry, index);
+                    })}
+
+                    {renderGroupTitle('quotes', groups.quotes.length > 0)}
+                    {groups.quotes.map((entry) => {
+                      const index = currentIndex++;
+                      return renderSearchItem(entry, index);
+                    })}
+
+                    {renderGroupTitle('credits', groups.credits.length > 0)}
+                    {groups.credits.map((entry) => {
+                      const index = currentIndex++;
+                      return renderSearchItem(entry, index);
+                    })}
+
+                    {renderGroupTitle('projects', groups.projects.length > 0)}
+                    {groups.projects.map((entry) => {
+                      const index = currentIndex++;
+                      return renderSearchItem(entry, index);
+                    })}
+
+                    {renderGroupTitle('tasks', groups.tasks.length > 0)}
+                    {groups.tasks.map((entry) => {
+                      const index = currentIndex++;
+                      return renderSearchItem(entry, index);
+                    })}
+
+                    {renderGroupTitle(
+                      'purchase_orders',
+                      groups.purchase_orders.length > 0
+                    )}
+                    {groups.purchase_orders.map((entry) => {
+                      const index = currentIndex++;
+                      return renderSearchItem(entry, index);
+                    })}
+
+                    {renderGroupTitle('settings', groups.settings.length > 0)}
+                    {groups.settings.map((entry) => {
+                      const index = currentIndex++;
+                      return renderSearchItem(entry, index);
+                    })}
+
+                    {groups.other.map((entry) => {
+                      const index = currentIndex++;
+                      return renderSearchItem(
+                        entry,
+                        index,
+                        options
+                          .toArray()
+                          .findIndex(
+                            (item) =>
+                              (item as Entry<SearchRecord>).id ===
+                              groups.other[0]?.id
+                          )
+                      );
+                    })}
+                  </>
+                );
+              })()}
```

Comment:
Is it possible to extract this into subcomponent and provide checks there? Feels kinda odd to place such a bit chunk of logic in the view part.

Reply:
@beganovich I agree, let's unburden the main Search component. I've done a nice refactoring to separate the logic into different components. Let me know your thoughts.